### PR TITLE
Rewrite history scavenger

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Rewrite history scavenger for maintenance
* Decrease page size reducing number of tombstones per DB call

**Why?**
N/A

**How did you test it?**
N/A

**Potential risks**
N/A

**Is hotfix candidate?**
N/A

### CODE REVIEW

File: common/persistence/cassandra/cassandraHistoryPersistence.go [hunk 1]
```diff
@@ -356,17 +356,17 @@ func (h *cassandraHistoryV2Persistence) DeleteHistoryBranch(
 	for _, b := range rsp.Branches {
 		for _, br := range b.Ancestors {
 			// These string casts are safe and optimized away -- https://github.com/golang/go/issues/3512
```
Comment (mastermanu) [inline common/persistence/cassandra/cassandraHistoryPersistence.go:358]: Is this comment on line 358 still applicable?

File: service/worker/scanner/history/scavenger.go [hunk 2]
```diff
@@ -105,186 +107,223 @@ func NewScavenger(
...
-	concurrency := s.rps/rpsPerConcurrency + 1
+	reqCh := make(chan taskDetail, pageSize)
 
-	for i := 0; i < concurrency; i++ {
-		go s.startTaskProcessor(ctx, taskCh, respCh)
+	for i := 0; i < numWorker; i++ {
+		s.WaitGroup.Add(1)
+		go s.taskWorker(ctx, reqCh)
 	}
+	s.WaitGroup.Add(1)
```
Comment (mastermanu) [inline service/worker/scanner/history/scavenger.go:131]: what value is this additional Add(1) providing? The task workers anyway won't exit until loadTasks finishes, right?

**Review by mastermanu — APPROVED** [review]

File: common/persistence/cassandra/cassandraHistoryPersistence.go (hunk 1 above)
Comment (wxing1292) [reply common/persistence/cassandra/cassandraHistoryPersistence.go:358]: i will update the comments in new PRs
```go
GetBranchId() string
```

File: service/worker/scanner/history/scavenger.go (hunk 2 above)
Comment (wxing1292) [reply service/worker/scanner/history/scavenger.go:131]: down below the logic do wait?
```go
s.WaitGroup.Wait()
```

**Pull request merged**